pages/login/LoginHandler.php

```diff
@@ -223,45 +223,41 @@ public function requestResetPassword($args, $request)
-                    'error' => 'user.login.lostPassword.confirmationSentFailedWithReason',
-                    'reason' => empty($reason = $user->getDisabledReason() ?? '')
-                        ? __('user.login.accountDisabled')
-                        : __('user.login.accountDisabledWithReason', ['reason' => htmlspecialchars($reason)])
-                ])
-                ->display('frontend/pages/userLostPassword.tpl');
+        if ($user !== null) {
+            
+            if ($user->getDisabled()) {
+                $templateMgr
+                    ->assign([
+                        'error' => 'user.login.lostPassword.confirmationSentFailedWithReason',
+                        'reason' => empty($reason = $user->getDisabledReason() ?? '')
+                            ? __('user.login.accountDisabled')
+                            : __('user.login.accountDisabledWithReason', ['reason' => htmlspecialchars($reason)])
+                    ])
+                    ->display('frontend/pages/userLostPassword.tpl');
+    
+                return;
+            }
 
-            return;
+            // Send email confirming password reset
+            $site = $request->getSite(); /** @var Site $site */
+            $context = $request->getContext(); /** @var Context $context */
+            $template = Repo::emailTemplate()->getByKey(
+                $context ? $context->getId() : PKPApplication::CONTEXT_SITE,
+                PasswordResetRequested::getEmailTemplateKey()
+            );
+            $mailable = (new PasswordResetRequested($site))
+                ->recipients($user)
+                ->from($site->getLocalizedContactEmail(), $site->getLocalizedContactName())
+                ->body($template->getLocalizedData('body'))
+                ->subject($template->getLocalizedData('subject'));
+            Mail::send($mailable);
+            
         }
 
-        // Send email confirming password reset
-        $site = $request->getSite(); /** @var Site $site */
-        $context = $request->getContext(); /** @var Context $context */
-        $template = Repo::emailTemplate()->getByKey(
-            $context ? $context->getId() : PKPApplication::CONTEXT_SITE,
-            PasswordResetRequested::getEmailTemplateKey()
-        );
-        $mailable = (new PasswordResetRequested($site))
-            ->recipients($user)
-            ->from($site->getLocalizedContactEmail(), $site->getLocalizedContactName())
-            ->body($template->getLocalizedData('body'))
-            ->subject($template->getLocalizedData('subject'));
-        Mail::send($mailable);
-
         $templateMgr->assign([
             'pageTitle' => 'user.login.resetPassword',
             'message' => 'user.login.lostPassword.confirmationSent',
-            'backLink' => $request->url(null, $request->getRequestedPage(), null, null, ['username' => $user->getUsername()]),
+            'backLink' => $request->url(null, $request->getRequestedPage(), null, null),
```

Comment:
This reverts https://github.com/pkp/pkp-lib/issues/9169. Since we're looking up the user by email address, and planning to add login by email, what do you think about moving ahead with merging https://github.com/pkp/pkp-lib/issues/8871 and using the email address in place of username here?

Reply:
Sorry for the delay in getting to this! Thinking this over from a higher level, I think allowing users to log in via email will solve the underlying problem, which is that usernames are easily forgotten and getting less common. I would be OK with reverting #9169 in the process of merging email-based logins.

Reply:
@asmecher PRs are done But note that reversion of https://github.com/pkp/pkp-lib/issues/9169 has not  implemented in `stable-3.3.0` as `stable-3.3.0` does not have login via email functionality . We only have login via email for `stable-3.4.0` and `main` .